src/__integrations__/docker/pmClient.pay3ds2.integration.docker.test.ts

```diff
@@ -0,0 +1,316 @@
+    ).fold(
+      _ => fail(),
+      res => res.value,
+    );
+
+    // approve terms
+    (
+      await pmClient.approveTermsUsingPOST({
+        Bearer: `Bearer ${startSessionResponse?.sessionToken}`,
+        approveTermsRequest: {
+          data: {
+            terms: true,
+            privacy: true,
+          },
+        },
+      })
+    ).fold(
+      () => fail(),
+      res => res.value?.data,
+    );
+
+    // POST Wallet
+    walletResponse = (
+      await pmClient.addWalletUsingPOST({
+        Bearer: `Bearer ${startSessionResponse?.sessionToken}`,
+        walletRequest: {
+          data: {
+            type: TypeEnum.CREDIT_CARD,
+            creditCard: {
+              brand: 'yyy',
+              expireMonth: '03',
+              expireYear: '25',
+              holder: 'UserName UserSurname',
+              pan: '5133765016844455',
+              securityCode: '666',
+            },
+            favourite: true,
+            idPagamentoFromEC: myIdPayment,
+          },
+        },
+        language: 'it',
+      })
+    ).fold(
+      () => fail(),
+      res => res.value?.data,
+    );
+  });
+
+  it('should return idWallet, idPayment and amount matching the ones in previous calls ', async () => {
+    // Pay
+    // Please note that the Pay request doesn't contain IP address and Browser Accepted Headers
+    const payResponse = (
+      await pmClient.pay3ds2UsingPOST({
+        Bearer: `Bearer ${startSessionResponse?.sessionToken}`,
+        id: myIdPayment,
+        payRequest: {
+          data: {
+            idWallet: walletResponse?.idWallet,
+            cvv: '666',
+            threeDSData: JSON.stringify({
+              acctId: `ACCT_${walletResponse.idWallet?.toString().trim()}`,
+              browserColorDepth: 30,
+              browserJavaEnabled: false,
+              browserLanguage: 'it-IT',
+              browserScreenHeight: 1120,
+              browserScreenWidth: 1792,
+              browserTZ: -60,
+              browserUserAgent:
+                'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36',
+              deliveryEmailAddress: 'username@domain.us',
+              workPhone: '3336666666',
+            }),
+          },
+        },
+        language: 'it',
+      })
+    ).fold(
+      () => fail(),
+      res => res.value?.data,
+    );
+
+    expect(payResponse?.idWallet).toEqual(walletResponse?.idWallet);
+    expect(payResponse?.nodoIdPayment).toEqual(myIdPayment);
+    expect(fromNullable(payResponse?.amount?.amount).getOrElse(0)).toEqual(
+      fromNullable(checkResponse?.amount?.amount).getOrElse(0),
+    );
+    expect(
+      (fromNullable(payResponse?.amount?.amount).getOrElse(0) as number) +
+        (fromNullable(payResponse?.fee?.amount).getOrElse(0) as number),
+    ).toEqual(fromNullable(payResponse?.grandTotal?.amount).getOrElse(0));
+  });
+
+  it('should execute STEP0 of 3ds2 flow', async () => {
```

Comment:
I'd change it in to something like `should check first 3ds2 transcaction step - TX_WAIT_FOR_3DS2_ACS_METHOD`